mono/metadata/object.c

```diff
@@ -6946,37 +6946,38 @@ mono_value_box_handle (MonoDomain *domain, MonoClass *klass, gpointer value, Mon
 	return_val_if_nok (error, NULL_HANDLE);
 
 	size -= MONO_ABI_SIZEOF (MonoObject);
-
-	if (mono_gc_is_moving ()) {
-		g_assert (size == mono_class_value_size (klass, NULL));
-		MONO_ENTER_NO_SAFEPOINTS;
-		gpointer data = mono_handle_get_data_unsafe (res_handle);
-		mono_gc_wbarrier_value_copy_internal (data, value, 1, klass);
-		MONO_EXIT_NO_SAFEPOINTS;
-	} else {
-		MONO_ENTER_NO_SAFEPOINTS;
-		gpointer data = mono_handle_get_data_unsafe (res_handle);
-#if NO_UNALIGNED_ACCESS
-		mono_gc_memmove_atomic (data, value, size);
-#else
-		switch (size) {
-		case 1:
-			*(guint8*)data = *(guint8 *) value;
-			break;
-		case 2:
-			*(guint16 *)(data) = *(guint16 *) value;
-			break;
-		case 4:
-			*(guint32 *)(data) = *(guint32 *) value;
-			break;
-		case 8:
-			*(guint64 *)(data) = *(guint64 *) value;
-			break;
-		default:
+	if (value != NULL) { 
```

Comment:
So if the source value is null we return a result with some default contents?  That seems easy to use incorrectly.  

Can we assert in here that `value != NULL` and fix the callers instead?  or maybe there is some managed exception that we should throw?